File: meet-guide-backend/src/services/pronunciationService.js
/**
 * Pronunciation Service - Handles pronunciation data from mispronunciation-detection-system
 */
const fs = require('fs').promises;
const path = require('path');
const PronunciationFeedback = require('../models/PronunciationFeedback');
const User = require('../models/User');
const Meeting = require('../models/Meeting');

// Path to mispronunciation detection system
const MISPRONUNCIATION_PATH = path.resolve(__dirname, '../../../meet-guide-components/mispronunciation-detection-system');

/**
 * Get user's pronunciation feedback
 */
async function getUserFeedback(userId, meetingId = null) {
    const query = { user_id: userId };
    if (meetingId) {
        query.meeting_id = meetingId;
    }
    return PronunciationFeedback.find(query).sort({ processed_at: -1 });
}

/**
 * Get user's pronunciation summary
 */
async function getUserSummary(userId) {
    const feedbackList = await getUserFeedback(userId);
    const user = await User.findById(userId);
    
    const totalWords = feedbackList.reduce((sum, f) => sum + (f.total_words || 0), 0);
    const totalErrors = feedbackList.reduce((sum, f) => sum + (f.mispronounced_count || 0), 0);
    const avgAccuracy = feedbackList.length > 0 
        ? feedbackList.reduce((sum, f) => sum + (f.accuracy || 0), 0) / feedbackList.length 
        : 100;

    // Calculate common errors
    const errorCounts = {};
    feedbackList.forEach(feedback => {
        if (feedback.errors) {
            feedback.errors.forEach(err => {
                const word = (err.word || err.spoken || '').toLowerCase();
                if (word) {
                    errorCounts[word] = (errorCounts[word] || 0) + 1;
                }
            });
        }
    });

    const commonErrors = Object.entries(errorCounts)
        .map(([word, count]) => ({ word, count }))
        .sort((a, b) => b.count - a.count)
        .slice(0, 10);

    return {
        user_id: userId,
        user_name: user ? user.username : 'Unknown',
        total_meetings: feedbackList.length,
        total_words: totalWords,
        total_errors: totalErrors,
        average_accuracy: avgAccuracy,
        common_errors: commonErrors,
        recent_feedback: feedbackList.slice(0, 5).map(f => ({
            meeting_id: f.meeting_id,
            total_words: f.total_words,
            errors: f.mispronounced_count,
            accuracy: f.accuracy,
            date: f.processed_at
        }))
    };
}

/**
 * Get meeting pronunciation feedback
 */
async function getMeetingFeedback(meetingId) {
    return PronunciationFeedback.find({ meeting_id: meetingId });
}

/**
 * Load pronunciation summary from files
 */
async function loadPronunciationSummary(recordingFolder) {
    try {
        const summaryPath = path.join(MISPRONUNCIATION_PATH, recordingFolder, 'participant_transcripts', 'mispronunciation_summary.json');
        const data = await fs.readFile(summaryPath, 'utf8');
        return JSON.parse(data);
    } catch (error) {
        return null;
    }
}

/**
 * Load timeline data
 */
async function loadTimelineData(recordingFolder) {
    try {
        const timelinePath = path.join(MISPRONUNCIATION_PATH, recordingFolder, 'timeline.json');
        const data = await fs.readFile(timelinePath, 'utf8');
        return JSON.parse(data);
    } catch (error) {
        return null;
    }
}

/**
 * Get participant mispronunciation detail
 */
async function getParticipantMispronunciation(recordingFolder, participantName) {
    const variations = [
        participantName,
        participantName.charAt(0).toUpperCase() + participantName.slice(1).toLowerCase(),
        participantName.toLowerCase(),
        participantName.toUpperCase()
    ];

    for (const name of variations) {
        try {
            const detailPath = path.join(MISPRONUNCIATION_PATH, recordingFolder, 'participant_transcripts', `${name}_mispronunciation.json`);
            const data = await fs.readFile(detailPath, 'utf8');
            return JSON.parse(data);
        } catch (e) {
            // Try next variation
        }
    }
    return null;
}

/**
 * Get participant transcript
 */
async function getParticipantTranscript(recordingFolder, participantName) {
    const variations = [
        participantName,
        participantName.charAt(0).toUpperCase() + participantName.slice(1).toLowerCase(),
        participantName.toLowerCase(),
        participantName.toUpperCase()
    ];

    for (const name of variations) {
        try {
            const transcriptPath = path.join(MISPRONUNCIATION_PATH, recordingFolder, 'participant_transcripts', `${name}.txt`);
            return await fs.readFile(transcriptPath, 'utf8');
        } catch (e) {
            // Try next variation
        }
    }
    return null;
}

/**
 * Get available meetings with pronunciation data
 */
async function getAvailableMeetings() {
    const meetings = [];
    
    try {
        const folders = await fs.readdir(MISPRONUNCIATION_PATH);
        
        for (const folder of folders) {
            if (folder.startsWith('.') || ['web', '__pycache__', 'configs', 'finetuned_whisper_nptel'].includes(folder)) {
                continue;
            }
            
            const folderPath = path.join(MISPRONUNCIATION_PATH, folder);
            
            try {
                const stat = await fs.stat(folderPath);
                if (!stat.isDirectory()) continue;
                
                const summaryPath = path.join(folderPath, 'participant_transcripts', 'mispronunciation_summary.json');
                
                try {
                    await fs.access(summaryPath);
                    
                    const summaryData = await loadPronunciationSummary(folder);
                    const timeline = await loadTimelineData(folder);
                    
                    meetings.push({
                        folder: folder,
                        name: folder.replace(/_/g, ' ').replace(/projectmeeting/i, 'Project Meeting '),
                        has_data: true,
                        participant_count: summaryData ? summaryData.total_participants : 0,
                        total_words: summaryData?.overall_stats?.total_words || 0,
                        total_errors: summaryData?.overall_stats?.total_errors || 0,
                        average_accuracy: summaryData?.overall_stats?.average_accuracy || 0,
                        meeting_date: timeline?.meeting_start_iso || null,
                        duration_sec: timeline?.total_duration_sec || null
                    });
                } catch (e) {
                    // No summary file
                }
            } catch (e) {
                // Not a valid folder
            }
        }
    } catch (error) {
        console.error('Error getting available meetings:', error.message);
    }
    
    return meetings;
}

/**
 * Import meeting data from projectmeeting folder
 */
async function importMeetingData(recordingFolder) {
    try {
        console.log(`Importing meeting data from: ${recordingFolder}`);
        
        const summaryData = await loadPronunciationSummary(recordingFolder);
        if (!summaryData) {
            return { success: false, message: 'No pronunciation data found' };
        }

        const timeline = await loadTimelineData(recordingFolder);
        const participants = summaryData.participants || {};
        let importedCount = 0;
        const importedUsers = [];
        const bcrypt = require('bcryptjs');

        // Create or update meeting record
        let meeting = await Meeting.findOne({ meeting_id: recordingFolder });
        const firstUser = await User.findOne({});
        
        if (!meeting) {
            // Build proper participants array with subdocuments
            const participantDocs = [];
            for (const pName of Object.keys(participants)) {
                const pLower = pName.toLowerCase();
                const pEmail = pLower.includes('@') ? pLower : `${pLower}@gmail.com`;
                const pUser = pLower.replace(/@.*$/, '');
                participantDocs.push({
                    username: pUser,
                    email: pEmail,
                    full_name: pName,
                    joined_at: new Date()
                });
            }

            meeting = new Meeting({
                meeting_id: recordingFolder,
                title: recordingFolder.replace(/_/g, ' ').replace(/projectmeeting/i, 'Project Meeting '),
                description: `Imported meeting with ${summaryData.total_participants} participants`,
                host_id: firstUser ? firstUser._id : null,
                host_name: firstUser ? firstUser.username : 'Unknown',
                host_email: firstUser ? firstUser.email : '',
                participants: participantDocs,
                status: 'ended',
                recording_folder: recordingFolder,
                actual_start: timeline?.meeting_start_iso ? new Date(timeline.meeting_start_iso) : new Date(),
                actual_end: timeline?.meeting_start_iso && timeline?.total_duration_sec 
                    ? new Date(new Date(timeline.meeting_start_iso).getTime() + parseFloat(timeline.total_duration_sec) * 1000)
                    : new Date()
            });
            await meeting.save();
            console.log(`Created meeting record for ${recordingFolder}`);
        }

        // Import pronunciation data for each participant
        for (const [participantName, pData] of Object.entries(participants)) {
            if (pData.status !== 'success') continue;

            // Determine if participantName is an email or plain name
            const isEmail = participantName.includes('@');
            const emailAddr = isEmail ? participantName.toLowerCase() : `${participantName.toLowerCase()}@gmail.com`;
            const userName = isEmail ? participantName.split('@')[0].toLowerCase() : participantName.toLowerCase();

            // Find user by email or username (case insensitive)
            let user = await User.findOne({
                $or: [
                    { email: { $regex: new RegExp(`^${emailAddr.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}$`, 'i') } },
                    { username: { $regex: new RegExp(`^${userName.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}$`, 'i') } }
                ]
            });
            
            if (!user) {
                // Create user if not exists
                const salt = await bcrypt.genSalt(10);
                const hashedPassword = await bcrypt.hash('password123', salt);
                
                user = new User({
                    email: emailAddr,
                    username: userName,
                    full_name: participantName,
                    hashed_password: hashedPassword
                });
                await user.save();
                console.log(`Created user: ${emailAddr}`);
            }

            // Load detailed mispronunciation data
            const detailData = await getParticipantMispronunciation(recordingFolder, participantName);
            const transcript = await getParticipantTranscript(recordingFolder, participantName);

            const errors = [];
            if (detailData && detailData.errors) {
                for (const err of detailData.errors) {
                    errors.push({
                        word: err.word || '',
                        spoken: err.word || '',
                        expected: err.expected || err.word || '',
                        expected_phonemes: (err.expected_phonemes || []).join(' '),
                        actual_phonemes: '',
                        error_type: err.error_type || 'pronunciation',
                        severity: err.severity || 'medium',
                        confidence: err.confidence || 0,
                        start_time: err.start_time || 0,
                        end_time: err.end_time || 0,
                        context: err.context || '',
                        suggestion: err.suggestion || ''
                    });
                }
            }

            // Upsert pronunciation feedback
            // Always store user_name as the user's email for consistent lookups
            await PronunciationFeedback.findOneAndUpdate(
                { meeting_id: recordingFolder, user_id: user._id },
                {
                    meeting_id: recordingFolder,
                    user_id: user._id,
                    user_name: user.email,
                    total_words: pData.total_words || 0,
                    mispronounced_count: pData.errors_detected || 0,
                    accuracy: (pData.accuracy || 1) * 100,
                    error_rate: pData.errors_detected && pData.total_words 
                        ? (pData.errors_detected / pData.total_words) * 100 
                        : 0,
                    mispronunciations: errors,
                    transcript: transcript || '',
                    processed_at: new Date()
                },
                { upsert: true, new: true }
            );

            importedCount++;
            importedUsers.push(participantName);
        }

        return { 
            success: true, 
            message: `Imported ${importedCount} participants`, 
            imported_users: importedUsers,
            meeting_id: recordingFolder
        };
    } catch (error) {
        console.error('Error importing meeting data:', error);
        return { success: false, message: error.message };
    }
}

/**
 * Get complete meeting feedback with all participants
 * If userId is provided, only return feedback for that user
 */
async function getCompleteMeetingFeedback(meetingId, userId = null) {
    const feedbackQuery = { meeting_id: meetingId };
    if (userId) {
        feedbackQuery.user_id = userId;
    }
    const feedbackList = await PronunciationFeedback.find(feedbackQuery);
    const meeting = await Meeting.findOne({ meeting_id: meetingId });
    const summaryData = await loadPronunciationSummary(meetingId);
    const timeline = await loadTimelineData(meetingId);

    // Compute overall stats scoped to the filtered participants
    let overallStats = null;
    if (feedbackList.length > 0) {
        const totalWords = feedbackList.reduce((sum, f) => sum + (f.total_words || 0), 0);
        const totalErrors = feedbackList.reduce((sum, f) => sum + (f.mispronounced_count || 0), 0);
        const avgAccuracy = feedbackList.reduce((sum, f) => sum + (f.accuracy || 0), 0) / feedbackList.length;
        overallStats = {
            total_words: totalWords,
            total_errors: totalErrors,
            average_accuracy: avgAccuracy / 100
        };
    } else if (!userId) {
        overallStats = summaryData?.overall_stats || null;
    }

    return {
        meeting: meeting ? {
            id: meeting._id,
            meeting_id: meeting.meeting_id,
            title: meeting.title,
            status: meeting.status,
            start_time: meeting.actual_start,
            end_time: meeting.actual_end,
            host_name: meeting.host_name
        } : null,
        timeline: timeline ? {
            start_time: timeline.meeting_start_iso,
            duration_sec: timeline.total_duration_sec,
            event_count: timeline.event_count
        } : null,
        overall_stats: overallStats,
        participants: feedbackList.map(f => ({
            user_id: f.user_id,
            user_name: f.user_name,
            total_words: f.total_words,
            errors: f.mispronounced_count,
            accuracy: f.accuracy,
            error_rate: f.error_rate,
            mispronunciations: f.errors || [],
            transcript: f.transcript
        }))
    };
}

module.exports = {
    getUserFeedback,
    getUserSummary,
    getMeetingFeedback,
    loadPronunciationSummary,
    loadTimelineData,
    getParticipantMispronunciation,
    getParticipantTranscript,
    getAvailableMeetings,
    importMeetingData,
    getCompleteMeetingFeedback,
    MISPRONUNCIATION_PATH
};

File: meet-guide-backend/app/services/pronunciation_service.py
"""
Pronunciation Service - Handle pronunciation feedback operations
"""
from datetime import datetime
from typing import Optional, List, Dict, Any
from bson import ObjectId
import json
import os
from pathlib import Path
import subprocess
import logging

from ..database import get_pronunciation_feedback_collection, get_users_collection
from ..models import PronunciationFeedback, MispronunciationError, PronunciationSummary
from ..config import get_settings
from .user_service import UserService

logger = logging.getLogger(__name__)
settings = get_settings()


class PronunciationService:
    """Service for pronunciation feedback operations"""
    
    @staticmethod
    def get_mispronunciation_system_path() -> Path:
        """Get the path to the mispronunciation detection system"""
        return Path(settings.mispronunciation_system_path).resolve()
    
    @staticmethod
    def get_recordings_path() -> Path:
        """Get the path to recordings"""
        return Path(settings.recordings_base_path).resolve()
    
    @staticmethod
    async def process_meeting_pronunciation(meeting_id: str, recording_folder: str) -> bool:
        """
        Process pronunciation analysis for a meeting.
        This runs the mispronunciation detection pipeline.
        """
        try:
            system_path = PronunciationService.get_mispronunciation_system_path()
            recording_path = PronunciationService.get_recordings_path() / recording_folder
            
            # Check if recording folder exists in mispronunciation system
            target_path = system_path / recording_folder
            
            if not target_path.exists():
                # Copy or link recording folder to mispronunciation system
                logger.info(f"Recording folder not found in mispronunciation system: {target_path}")
                # For now, we'll assume the recordings are already in place or symlinked
            
            # Run the process_meeting.py script
            process_script = system_path / "process_meeting.py"
            
            if not process_script.exists():
                logger.error(f"Process script not found: {process_script}")
                return False
            
            logger.info(f"Processing pronunciation for meeting: {recording_folder}")
            
            result = subprocess.run(
                ["python", str(process_script), recording_folder],
                cwd=str(system_path),
                capture_output=True,
                text=True
            )
            
            if result.returncode != 0:
                logger.error(f"Pronunciation processing failed: {result.stderr}")
                return False
            
            logger.info(f"Pronunciation processing completed for: {recording_folder}")
            return True
            
        except Exception as e:
            logger.error(f"Error processing pronunciation: {e}")
            return False
    
    @staticmethod
    async def load_pronunciation_data(recording_folder: str) -> Optional[Dict[str, Any]]:
        """Load pronunciation data from the mispronunciation system output"""
        try:
            system_path = PronunciationService.get_mispronunciation_system_path()
            
            # Try new structure first
            summary_file = system_path / recording_folder / "participant_transcripts" / "mispronunciation_summary.json"
            
            if not summary_file.exists():
                # Try old structure
                summary_file = system_path / recording_folder / "output" / "participant_transcripts" / "mispronunciation_summary.json"
            
            if not summary_file.exists():
                logger.warning(f"Summary file not found for: {recording_folder}")
                return None
            
            with open(summary_file, 'r') as f:
                return json.load(f)
                
        except Exception as e:
            logger.error(f"Error loading pronunciation data: {e}")
            return None
    
    @staticmethod
    async def get_participant_pronunciation_detail(
        recording_folder: str, 
        participant_name: str
    ) -> Optional[Dict[str, Any]]:
        """Get detailed pronunciation data for a specific participant"""
        try:
            system_path = PronunciationService.get_mispronunciation_system_path()
            
            # Try new structure first
            detail_file = system_path / recording_folder / "participant_transcripts" / f"{participant_name}_mispronunciation.json"
            
            if not detail_file.exists():
                # Try with capitalized name
                detail_file = system_path / recording_folder / "participant_transcripts" / f"{participant_name.capitalize()}_mispronunciation.json"
            
            if not detail_file.exists():
                logger.warning(f"Detail file not found for: {participant_name}")
                return None
            
            with open(detail_file, 'r') as f:
                return json.load(f)
                
        except Exception as e:
            logger.error(f"Error loading participant detail: {e}")
            return None
    
    @staticmethod
    async def get_participant_transcript(
        recording_folder: str, 
        participant_name: str
    ) -> Optional[str]:
        """Get transcript for a specific participant"""
        try:
            system_path = PronunciationService.get_mispronunciation_system_path()
            
            transcript_file = system_path / recording_folder / "participant_transcripts" / f"{participant_name}.txt"
            
            if not transcript_file.exists():
                transcript_file = system_path / recording_folder / "participant_transcripts" / f"{participant_name.capitalize()}.txt"
            
            if not transcript_file.exists():
                return None
            
            with open(transcript_file, 'r') as f:
                return f.read()
                
        except Exception as e:
            logger.error(f"Error loading transcript: {e}")
            return None
    
    @staticmethod
    async def save_pronunciation_feedback(
        meeting_id: str,
        user_id: str,
        user_name: str,
        pronunciation_data: Dict[str, Any],
        transcript: Optional[str] = None
    ) -> PronunciationFeedback:
        """Save pronunciation feedback to database"""
        feedback_collection = get_pronunciation_feedback_collection()
        
        # Parse errors from pronunciation data
        errors = []
        for error in pronunciation_data.get("mispronunciations", []):
            errors.append(MispronunciationError(
                word=error.get("word", ""),
                expected_phonemes=error.get("expected_phonemes", ""),
                actual_phonemes=error.get("actual_phonemes"),
                error_type=error.get("error_type", "unknown"),
                severity=error.get("severity", "medium"),
                confidence=error.get("confidence", 0.0),
                timestamp=error.get("timestamp"),
                suggestion=error.get("suggestion")
            ))
        
        feedback = PronunciationFeedback(
            meeting_id=meeting_id,
            user_id=user_id,
            user_name=user_name,
            total_words=pronunciation_data.get("total_words", 0),
            mispronounced_count=pronunciation_data.get("mispronounced", len(errors)),
            error_rate=pronunciation_data.get("error_rate", 0.0),
            errors=errors,
            transcript=transcript,
            processed_at=datetime.utcnow()
        )
        
        # Upsert feedback
        feedback_doc = feedback.model_dump()
        feedback_doc["errors"] = [e.model_dump() for e in errors]
        
        result = await feedback_collection.update_one(
            {"meeting_id": meeting_id, "user_id": user_id},
            {"$set": feedback_doc},
            upsert=True
        )
        
        if result.upserted_id:
            feedback.id = str(result.upserted_id)
        
        logger.info(f"Saved pronunciation feedback for {user_name} in meeting {meeting_id}")
        return feedback
    
    @staticmethod
    async def get_user_pronunciation_feedback(
        user_id: str,
        meeting_id: Optional[str] = None
    ) -> List[PronunciationFeedback]:
        """Get pronunciation feedback for a user"""
        feedback_collection = get_pronunciation_feedback_collection()
        
        query = {"user_id": user_id}
        if meeting_id:
            query["meeting_id"] = meeting_id
        
        cursor = feedback_collection.find(query).sort("processed_at", -1)
        
        result = []
        async for doc in cursor:
            errors = [MispronunciationError(**e) for e in doc.get("errors", [])]
            result.append(PronunciationFeedback(
                id=str(doc["_id"]),
                meeting_id=doc["meeting_id"],
                user_id=doc["user_id"],
                user_name=doc["user_name"],
                total_words=doc.get("total_words", 0),
                mispronounced_count=doc.get("mispronounced_count", 0),
                error_rate=doc.get("error_rate", 0.0),
                errors=errors,
                transcript=doc.get("transcript"),
                processed_at=doc["processed_at"]
            ))
        
        return result
    
    @staticmethod
    async def get_meeting_pronunciation_feedback(meeting_id: str) -> List[PronunciationFeedback]:
        """Get all pronunciation feedback for a meeting"""
        feedback_collection = get_pronunciation_feedback_collection()
        
        cursor = feedback_collection.find({"meeting_id": meeting_id})
        
        result = []
        async for doc in cursor:
            errors = [MispronunciationError(**e) for e in doc.get("errors", [])]
            result.append(PronunciationFeedback(
                id=str(doc["_id"]),
                meeting_id=doc["meeting_id"],
                user_id=doc["user_id"],
                user_name=doc["user_name"],
                total_words=doc.get("total_words", 0),
                mispronounced_count=doc.get("mispronounced_count", 0),
                error_rate=doc.get("error_rate", 0.0),
                errors=errors,
                transcript=doc.get("transcript"),
                processed_at=doc["processed_at"]
            ))
        
        return result
    
    @staticmethod
    async def get_user_pronunciation_summary(user_id: str) -> PronunciationSummary:
        """Get summary of all pronunciation feedback for a user"""
        feedback_list = await PronunciationService.get_user_pronunciation_feedback(user_id)
        
        # Get user info
        user = await UserService.get_user_by_id(user_id)
        user_name = user.username if user else "Unknown"
        
        total_words = sum(f.total_words for f in feedback_list)
        total_errors = sum(f.mispronounced_count for f in feedback_list)
        avg_error_rate = (total_errors / total_words * 100) if total_words > 0 else 0
        
        # Calculate most common errors
        error_counts: Dict[str, int] = {}
        for feedback in feedback_list:
            for error in feedback.errors:
                word = error.word.lower()
                error_counts[word] = error_counts.get(word, 0) + 1
        
        most_common = sorted(
            [{"word": k, "count": v} for k, v in error_counts.items()],
            key=lambda x: x["count"],
            reverse=True
        )[:10]
        
        # Calculate improvement over time
        improvement = []
        for feedback in sorted(feedback_list, key=lambda f: f.processed_at):
            improvement.append({
                "date": feedback.processed_at.isoformat(),
                "meeting_id": feedback.meeting_id,
                "error_rate": feedback.error_rate
            })
        
        return PronunciationSummary(
            user_id=user_id,
            user_name=user_name,
            total_meetings=len(feedback_list),
            total_words_spoken=total_words,
            total_mispronunciations=total_errors,
            average_error_rate=avg_error_rate,
            most_common_errors=most_common,
            improvement_over_time=improvement
        )
    
    @staticmethod
    async def import_existing_meeting_data(recording_folder: str) -> bool:
        """
        Import pronunciation data from an existing processed meeting.
        Maps participants to users in the database.
        """
        try:
            # Load pronunciation summary
            summary_data = await PronunciationService.load_pronunciation_data(recording_folder)
            if not summary_data:
                logger.error(f"No pronunciation data found for: {recording_folder}")
                return False
            
            participants = summary_data.get("participants", {})
            
            for participant_name, p_data in participants.items():
                if p_data.get("status") != "success":
                    continue
                
                # Try to find user by username (case-insensitive)
                user = await UserService.get_user_by_username(participant_name)
                
                if not user:
                    logger.warning(f"No user found for participant: {participant_name}")
                    continue
                
                # Load detailed pronunciation data
                detail_data = await PronunciationService.get_participant_pronunciation_detail(
                    recording_folder, participant_name
                )
                
                if not detail_data:
                    # Use summary data instead
                    detail_data = {
                        "total_words": p_data.get("total_words", 0),
                        "mispronounced": p_data.get("errors_detected", 0),
                        "error_rate": p_data.get("error_rate", 0.0),
                        "mispronunciations": []
                    }
                
                # Load transcript
                transcript = await PronunciationService.get_participant_transcript(
                    recording_folder, participant_name
                )
                
                # Save feedback
                await PronunciationService.save_pronunciation_feedback(
                    meeting_id=recording_folder,
                    user_id=user.id,
                    user_name=participant_name,
                    pronunciation_data=detail_data,
                    transcript=transcript
                )
            
            logger.info(f"Imported pronunciation data for: {recording_folder}")
            return True
            
        except Exception as e:
            logger.error(f"Error importing meeting data: {e}")
            return False
